tests/sklearn/test_sklearn_models.py

```diff
@@ -663,9 +694,20 @@ def check_grid_search(model_class, x, y, scoring):
     with warnings.catch_warnings():
         # Sometimes, we miss convergence, which is not a problem for our test
         warnings.simplefilter("ignore", category=ConvergenceWarning)
+        warnings.simplefilter("ignore", category=UndefinedMetricWarning)
+
+        if get_model_name(model_class) == "KNeighborsClassifier" and scoring in [
+            "roc_auc",
+            "average_precision",
+        ]:
+            # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/3962
+            pytest.skip("Skipping predict_proba for KNN, doesn't work for now")
+
+        # pylint: disable=invalid-name
+        cv = 2 if get_model_name(model_class) == "KNeighborsClassifier" else 5
```

Comment:
I think cv is not that important. If 5 is too large let's just pick 2 or 3. We just want gridsearch to work with our models. So you can probably remove this condition

Reply:
ok